File: runtimejs/java/util/function/Predicate.js
//reference// java/lang/NullPointerException
//load// java/lang/Object
var java_util_function_Predicate =  
{   _superinterfaces: [], 
    _defaults:
    {   and_1: function(other)
        {   if (other===null)
            {   throw (new java_lang_NullPointerException())._0()._e;
            }
            return new java_util_function_PredicateAnd(this,other);
        }
        ,
        negate_0: function()
        {   return new java_util_function_PredicateNegate(this);
        },
        or_1: function(other)
        {   if (other===null)
            {   throw (new java_lang_NullPointerException())._0()._e;
            }
            return new java_util_function_PredicateOr(this, other);
        }
    },
}; 

// methods:
// default Predicate<T> and(Predicate<T> other)
// static Predicate<T> isEqual(Object targetref)
// default Predicate<T> negate()
// default Predicate<T> or(Predicate<T> other)
// boolean test(T t)


var java_util_function_PredicateAnd = function(a,b)   
// internal use only - integrate allocator with constructor 
{   this.a = a;
    this.b = b;    
};
_class(java_util_function_PredicateAnd, java_lang_Object, [java_util_function_Predicate], 
"java.util.function.PredicateAnd"  //replace-me-with-empty-string-for-production//
,{  test_1: function(o) 
    {   return this.a.test_1(o) && this.b.test_1(o);
    },        
});

var java_util_function_PredicateNegate = function(a)
 // internal use only - integrate allocator with constructor 
{   this.a = a;
};
_class(java_util_function_PredicateNegate, java_lang_Object, [java_util_function_Predicate], 
"java.util.function.PredicateNegate"  //replace-me-with-empty-string-for-production//
,{  test_1: function(o) 
    {   return ! this.a.test_1(o);
    },        
});

var java_util_function_PredicateOr = function(a,b) 
 // internal use only - integrate allocator with constructor 
{   this.a = a;
    this.b = b;    
};
_class(java_util_function_PredicateOr, java_lang_Object, [java_util_function_Predicate], 
"java.util.function.PredicateOr"  //replace-me-with-empty-string-for-production//
,{   test_1: function(o) 
    {   return this.a.test_1(o) || this.b.test_1(o);
    },        
});

var java_util_function_PredicateIsEqual = function(tobj) 
  // internal use only - integrate allocator with constructor 
{   this.tobj = tobj;
};
_class(java_util_function_PredicateIsEqual, java_lang_Object, [java_util_function_Predicate], 
"java.util.function.PredicateIsEqual"  //replace-me-with-empty-string-for-production//
,{   test_1: function(o) 
    {   return this.tobj===null ? o===null : this.tobj.equals_1(o);
    },        
});

var java_util_function_Predicate_isEqual_1 = function(tobj) 
{   return new java_util_function_PredicateIsEqual(tobj);
};

File: runtimecs/java/util/function/Predicate.cs
using java.lang;

namespace java.util.function 
{ 
    public interface Predicate
    {
        Predicate and(Predicate other);
        Predicate negate();
        Predicate or(Predicate other);
        bool test(object s);
    }

    public static class Predicate_0009
    {
        public static Predicate and(Predicate @this, Predicate other)
        {   
            if (@this==null || other==null) { throw new NullPointerException(); }
            return new PredicateAnd(@this,other);
        }
        
        public static Predicate negate(Predicate @this)
        {   
            if (@this==null) { throw new NullPointerException(); }
            return new PredicateNegate(@this);
        }
        
        public static Predicate or(Predicate @this, Predicate other)
        {   
            if (@this==null || other==null) { throw new NullPointerException(); }
            return new PredicateOr(@this,other);
        }
        
        public static Predicate isEqual(object tobj) 
        {   
            return new PredicateIsEqual(tobj);
        }
    }

    public class PredicateIsEqual : Predicate
    {
        private readonly object tobj;
        
        public PredicateIsEqual(object tobj) 
        {   
            this.tobj = tobj;
        }
        public virtual bool test(object o)
        {   
            return tobj==null ? o==null : tobj.Equals(o);
        }
        public virtual Predicate and(Predicate other)
        {   
            return Predicate_0009.and(this,other);
        }
        public virtual Predicate negate()
        {   
            return Predicate_0009.negate(this);
        }
        public virtual Predicate or(Predicate other)
        {   
            return Predicate_0009.or(this,other);
        }
    }        

    public class PredicateAnd : PredicateIsEqual
    {
        private readonly Predicate a;
        private readonly Predicate b;
        
        public PredicateAnd(Predicate a, Predicate b) : base(null)
        {   
            this.a = a;
            this.b = b;
        }
        
        public override bool test(object o)
        {   
            return a.test(o) && b.test(o);
        }
    }        

    public class PredicateNegate : PredicateIsEqual
    {
        private readonly Predicate a;
        
        public PredicateNegate(Predicate a) : base(null)
        {   
            this.a = a;
        }
        public override bool test(object o)
        {   
            return !a.test(o);
        }
    }
            

    public class PredicateOr : PredicateIsEqual
    {
        private readonly Predicate a;
        private readonly Predicate b;
        
        public PredicateOr(Predicate a, Predicate b) : base(null)
        {   
            this.a = a;
            this.b = b;
        }
        public override bool test(object o)
        {   
            return a.test(o) || b.test(o);
        }
    }
}        
